app/controllers/invite_codes_controller.rb

```diff
@@ -15,4 +16,8 @@ def create
     def ensure_self_hosted
       redirect_to root_path unless self_hosted?
     end
+
+    def ensure_admin
+      redirect_to settings_hosting_path, alert: "You are not allowed to generate invite codes" unless Current.user.admin?
+    end
```

Comment:
I think since we're disabling the input in the UI, we should probably just throw an exception here.  A user should not be able to trigger this POST request given the UI, so if they try to side-step that, they should receive an error I think.